### ISSUE TITLE
Test sim_mutations with unary nodes

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#2137]
Just to be sure, we should add a quick test that checks that sim_mutations works OK when we have unary nodes (probably already exists for full_arg, but no harm in adding another test).

**Comment by General-Solution** [comment]
It doesn't seem like there is one for full_arg in test_algorithms.py so far. Should the test check that mutations can be simulated on a tree sequence with unary nodes, or should it also check that mutations show up in the same places as the non-unary case?

**Comment by jeromekelleher** [comment]
It's not ``test_algorithms`` we're interested in here I think (that's just the prototyper). I think we just want to have a quick test in ``test_mutations.py`` that checks it works on different topologies, e.g.:
```python

class TestNonStandardTopologies:
     def test_full_arg(self):
           ts = # simulate full ARG
           ts = msprime.sim_mutations(ts, ...)
           assert ts.num_mutations > 0
    
    def test_unary_nodes(self):
         ts = # simulate with unary nodes
         # Same as above
```

These are just really simple unit tests to make sure that (a) we don't fail or throw an error for these topologies; and (b) that we produce at least one mutation.

The statistical tests will check that what we're producing actually makes sense.

**Comment by GertjanBisschop** [comment]
Solved in #2152.
